rotkehlchen/chain/evm/decoding/hop/decoder.py

```diff
@@ -0,0 +1,259 @@
+        """This function is used to decode the TransferSentToL2 events on Hop protocol."""
+        recipient = hex_or_bytes_to_address(context.tx_log.topics[3])
+        if not self.base.is_tracked(recipient):
+            return DEFAULT_DECODING_OUTPUT
+
+        if (bridge := self.bridges.get(context.tx_log.address)) is None:
+            return DEFAULT_DECODING_OUTPUT
+
+        chain_id_raw = hex_or_bytes_to_int(context.tx_log.topics[2])
+        chain_id = ChainID.deserialize(chain_id_raw)
+        amount_raw = hex_or_bytes_to_int(context.tx_log.data[:32])
+        amount = self._get_amount(amount_raw, bridge)
+        bonder_fee_raw = hex_or_bytes_to_int(context.tx_log.data[64:96])
+        bonder_fee = token_normalized_value_decimals(bonder_fee_raw, DEFAULT_TOKEN_DECIMALS)
+
+        for event in context.decoded_events:
+            if (
+                event.address in (bridge.amm_wrapper, bridge.address) and
+                event.asset == bridge.asset and
+                event.event_type == HistoryEventType.SPEND and
+                event.event_subtype == HistoryEventSubType.NONE
+            ):
+                event.balance.amount = FVal(amount) - FVal(bonder_fee)
+                event.event_type = HistoryEventType.DEPOSIT
+                event.event_subtype = HistoryEventSubType.BRIDGE
+                event.counterparty = CPT_HOP
+                event.notes = f'Bridge {event.balance.amount} {event.asset.symbol_or_name()} to {chain_id.label()} via Hop protocol'  # noqa: E501
+
+                if bonder_fee > ZERO:
+                    fee_event = self.base.make_event_next_index(
+                        tx_hash=event.tx_hash,
+                        timestamp=ts_ms_to_sec(event.timestamp),
+                        event_type=HistoryEventType.SPEND,
+                        event_subtype=HistoryEventSubType.FEE,
+                        asset=event.asset,
+                        balance=Balance(amount=bonder_fee),
+                        location_label=event.location_label,
+                        notes=f'Spend {bonder_fee} {event.asset.symbol_or_name()} as a hop fee',
+                        counterparty=CPT_HOP,
+                        address=event.address,
+                    )
+                    context.decoded_events.append(fee_event)
+                    maybe_reshuffle_events(
+                        ordered_events=[event, fee_event],
+                        events_list=context.decoded_events,
+                    )
+                break
+        else:
+            # send method works differently than normal bridge events
+            for event in context.decoded_events:
+                if (
+                    event.address == ZERO_ADDRESS and
+                    event.asset == bridge.hop_asset and
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
```

Comment:
why loop again? Why don't make an `elif statement` checking for the zero address?